src/libYARP_sig/src/ImageFile.cpp

```diff
@@ -250,6 +254,93 @@ static bool ImageReadMono(ImageOf<PixelMono> &img, const char *filename)
 // private write methods
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+#if YARP_HAS_PNG_C
+static bool SavePNG(char *src, const char *filename, size_t h, size_t w, size_t rowSize, png_byte color_type, png_byte bit_depth)
+{
```

Comment:
`src` and `filename` are used but not checked to be `!=nullptr`, I'd like to add this check.

Reply:
Unfortunately this is not checked in the other similar methods as well. I'd start by adding the check here, and later fix the other methods

Reply:
libpng is a C library. Its APIs do not allow neither the use of smart pointers, nor fstream.